osquery/config/parsers/tests/options_tests.cpp

```diff
@@ -60,4 +60,20 @@ TEST_F(OptionsConfigParserPluginTests, test_unknown_option) {
   EXPECT_EQ(1U, doc["custom_fake"].GetUint());
   EXPECT_FALSE(Flag::getValue("custom_fake").empty());
 }
+
+TEST_F(OptionsConfigParserPluginTests, test_json_option) {
+  Config c;
+  std::map<std::string, std::string> update;
+
+  update["awesome"] =
+          "{\"options\": {\"complex\": {\"foo\": 1, \"bar\": \"baz\"}}}";
+  auto s = c.update(update);
+
+  const auto& doc = c.getParser("options")->getData().doc()["options"];
```

Comment:
I think it would be good idea to check `status` of `update` here. It could fail for some reason.

Reply:
done